src/client.rs

```diff
@@ -235,13 +235,31 @@ impl Spotify {
     /// - limit  - the number of albums to return
     /// - offset - the index of the first album to return
     ///
+    /// See [`Spotify::artist_albums_manual`] for a manually paginated version
+    /// of this.
+    ///
     /// [Reference](https://developer.spotify.com/documentation/web-api/reference/#endpoint-get-an-artists-albums)
+    pub fn artist_albums<'a>(
+        &'a self,
+        artist_id: &'a ArtistId,
+        album_type: Option<AlbumType>,
+        market: Option<&'a Market>,
+    ) -> impl Paginator<ClientResult<SimplifiedAlbum>> + 'a {
+        paginate(
+            move |limit, offset| {
+                self.artist_albums_manual(artist_id, album_type, market, Some(limit), Some(offset))
+            },
+            50,
+        )
+    }
+
+    /// The manually paginated version of [`Spotify::artist_albums`].
     #[maybe_async]
-    pub async fn artist_albums(
+    pub async fn artist_albums_manual(
         &self,
         artist_id: &ArtistId,
         album_type: Option<AlbumType>,
-        market: Option<Market>,
+        market: Option<&Market>,
```

Comment:
I've had to make some parameters a reference, like `Market`, because when using automatic pagination we iterate multiple times and would have to clone the parameter.

Reply:
Yes, it makes sense. As I mentioned in another PR, it might be a chance to pass parameters by reference